stdlib/public/core/ErrorType.swift

```diff
@@ -175,6 +175,10 @@ internal func _getErrorDefaultUserInfo<T: Error>(_ error: T) -> AnyObject?
 public func _bridgeErrorToNSError(_ error: __owned Error) -> AnyObject
 #endif
 
+/// SPI for registering a callback to be called whenever an error is thrown.
+@_silgen_name("_swift_addErrorWillThrowCallback")
+public func _addErrorWillThrowCallback(_ callback: @escaping (Error) -> Void)
```

Comment:
Don't have a strong opinion, but maybe `_addWillThrowErrorCallback` would be a clearer name? (i.e. "will throw error" instead of "error will throw".) Just since Error types don't perform the throw, they are thrown.